[MISC] Move core/pod_tuple.hpp -> utility/tuple/pod_tuple.hpp

eaasna opened the pull request:
Part of [seqan/product_backlog#160](https://github.com/seqan/product_backlog/issues/160)

- Moves core/pod_tuple.hpp -> utility/tuple/pod_tuple.hpp
- Updates ingroup to utility
- Deprecates core/pod_tuple.hpp
- Fixes include paths
- Moves pod_tuple_test.hpp and pod_tuple_snippet according to new directory structure
- Adjusts CMakeLists.txt files

MitraDarja commented on include/seqan3/core/all.hpp:
```diff
@@ -24,7 +24,7 @@
 #include <seqan3/core/detail/all.hpp>
 #include <seqan3/core/platform.hpp>
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
```
Alphabetical order?

MitraDarja commented on include/seqan3/core/tuple_utility.hpp:
```diff
@@ -14,7 +14,7 @@
 #include <utility>
 
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
 #include <seqan3/utility/tuple/concept.hpp>
```
Alphabetical order

MitraDarja commented on test/snippet/utility/tuple/pod_tuple.cpp:
```diff
@@ -1,4 +1,4 @@
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
 #include <seqan3/core/debug_stream.hpp>
```
Alphabetical order

eaasna replied:
Fixed the include order in the files you mentioned. Please review again :)

MitraDarja reviewed (changes requested):
Looks good, just one minor thing, headers are ordered alphabetically. If the order is at the moment incorrect, but will be correct once another file is moved from core to utility I think you can leave the incorrect order (I didn't mark this), all other of my marks should be reordered if I am not mistaken. :)

eseiler commented on include/seqan3/core/pod_tuple.hpp:
```diff
@@ -6,371 +6,12 @@
-
-} // namespace std
+SEQAN3_DEPRECATED_HEADER(
+"This header is deprecated and will be removed in SeqAn-3.1.0 Please #include <seqan3/utility/tuple/pod_tuple.hpp> instead.")
```
You are missing a new line here (files always end with a new line). All editor usually have a setting for this somewhere :)

eaasna replied:
Fixed it! Thanks for noticing :)

eaasna commented:
I fixed the merge conflicts and rebased my branch so this should now be good to go :) 

I also went back and cleaned up the commit history. The seqan contributor guide said to keep the number of commits minimal, wondering if it's wise to have just one bigger commit in this case?

MitraDarja reviewed (approved):
LGTM. Thanks. 👍

marehr commented on test/unit/utility/tuple_utility_test.cpp:
```diff
@@ -5,19 +5,16 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
-#include <seqan3/std/concepts>
-#include <tuple>
-#include <type_traits>
-
```
```suggestion
#include <seqan3/std/concepts>
#include <tuple>
#include <type_traits>

```

marehr commented on test/unit/utility/tuple_utility_test.cpp:
```diff
@@ -5,19 +5,16 @@
 
 #include <seqan3/core/detail/strong_type.hpp>
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/std/concepts>
+#include <tuple>
+#include <type_traits>
```
```suggestion
```

marehr reviewed (commented):
Thank you!

marehr reviewed (approved):
I committed the suggested changes for you! 

Thank you for the PR!

I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.

eaasna commented:
> I committed the suggested changes for you!
> 
> Thank you for the PR!
> 
> I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.

Thanks, but what is CI? @marehr

smehringer commented:
> > I committed the suggested changes for you!
> > Thank you for the PR!
> > I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.
> 
> Thanks, but what is CI? @marehr

CI is the abbreviation for *Continous Integration*, referring to the tests/checks that are running in every PR.
